src/arena.rs

```diff
@@ -187,6 +187,12 @@ impl ArenaHeader {
 #[derive(Debug)]
 pub struct TypedArenaPtr<T: ?Sized>(ptr::NonNull<T>);
 
+impl TypedArenaPtr<Integer> {
+    pub fn value(&self) -> Integer {
+        unsafe { self.0.as_ref().clone() }
+    }
+}
+
```

Comment:
This function isn't necessary. The `TypedArenaPtr<Integer>` can be dereferenced and the reference cloned.